docs/packaging_policy.md

```diff
@@ -20,3 +25,59 @@ release/download webpage.
 
 - In case of needing those binaries to use them as a "build require" for some library, we will consider following the approach of adding it
   as a system recipe (`<build_require>/system`) and making those binaries available in the CI machines (if the license allows it).
+
+
+## Settings
+
+All recipes should list the four settings `os`, `arch`, `compiler` and `build_type` so Conan will compute a different package ID
+for each combination. There are some particular cases for this general rule:
+ * **Recipes for _header only_ libraries** might omit the `settings` attibute, but in any case they should add
+  
+   ```python
+   def package_id(self):
+      self.info.header_only()
+   ```
+
+ * **Recipes that provide applications** (`b2`, `cmake`, `make`,...) that are generally used as a _build requires_, must list all
+   the settings as well, but they should remove the `compiler` one in the corresponding method unless the recipe provides also
```

Comment:
this statement needs to be checked carefully. it might be too strong a statement, confusing newcomers.
probably, compiler setting cannot be always safely removed, especially for Visual Studio, where DLL like `msvcrt<version>.dll` is linked to the executable, resulting in different binaries. or it needs a special treatment to ensure static runtime (MT) is used.
on linux, perhaps at least `compiler.libcxx` sub-setting matters, introducing a reference to `libstdc++.so` or `libc++.so`, in case of C++ builds.
it can be safely removed for many tools like `meson` or `automake` where are just scripts and no binaries.

Reply:
I'm only trying to document what was decided around this issue: https://github.com/conan-io/conan-center-index/issues/3157. Of course, there are runtime dependencies that we are not taking into account and are not modeled in the Conan settings.

For these reasons we try to use the _most_ compatible runtime deps possible, that's why we use Ubuntu 16.04 in Linux and we build the configurations in order, providing the binaries with the oldest compiler always (in Linux, Windows or Mac).

Windows binaries require redistributables for VS2017, and Linux ones require glibc from Ubuntu 16.04. So far, these requirements should be satisfied in 99% of the machines that are consuming Conan packages and working on C++ projects, don't you think so?